src/Nethermind/Nethermind.Evm/TransactionProcessing/TransactionProcessor.cs

```diff
@@ -404,16 +415,17 @@ protected virtual TransactionResult IncrementNonce(Transaction tx, BlockHeader h
             Transaction tx,
             in BlockExecutionContext blCtx,
             IReleaseSpec spec,
-            in UInt256 effectiveGasPrice)
+            in UInt256 effectiveGasPrice,
+            AuthorizedCodeInfoRepository authorizedCode)
         {
             Address recipient = tx.GetRecipient(tx.IsContractCreation ? WorldState.GetNonce(tx.SenderAddress) : 0);
             if (recipient is null) ThrowInvalidDataException("Recipient has not been resolved properly before tx execution");
 
-            TxExecutionContext executionContext = new(in blCtx, tx.SenderAddress, effectiveGasPrice, tx.BlobVersionedHashes);
+            TxExecutionContext executionContext = new(in blCtx, tx.SenderAddress, effectiveGasPrice, tx.BlobVersionedHashes, authorizedCode);
 
             CodeInfo codeInfo = tx.IsContractCreation
                 ? new(tx.Data ?? Memory<byte>.Empty)
-                : _codeInfoRepository.GetCachedCodeInfo(WorldState, recipient, spec);
+                : _authorizedCodeInfoRepository.GetCachedCodeInfo(WorldState, recipient, spec);
```

Comment:
Usage of _authorizedCodeRepository vs _codeRepository  and as field vs as param is very confusing.

Can we make it:
1. If EIP is enabled set coderepository to Authorized one else set it to normal one
2. have it initialize in one place and use everywhere
3. Authorized one would decorate normal one, if InsertCode is called it will relate it to normal one